cmd/bucket-metadata.go

```diff
@@ -373,3 +376,44 @@ func deleteBucketMetadata(ctx context.Context, obj objectDeleter, bucket string)
 	}
 	return nil
 }
+
+// migrate config for remote targets by encrypting data if currently unencrypted
+func (b *BucketMetadata) migrateTargetConfig(ctx context.Context, objectAPI ObjectLayer) error {
```

Comment:
Two problems with the same solution

First of all `utf8.Valid` is not sufficient as encrypted output may still validate as valid utf8. Not going to do the statistics, but is can happen.

Second of all this is rather costly just to migrate once.

I propose you add an identifier as the first byte.

If the existing content is JSON, adding a `0x01` to identify it as encrypted is safe since that will never be the first JSON byte. That way you only have to check that one byte.

Reply:
switched to encrypting target only if kms is configured. presence of encryption metadata makes it easier to know whether to migrate.